src/cmd/proxy/proxy.go

```diff
@@ -199,7 +199,10 @@ func (h *FlowAPIService) GetLatestBlockHeader(context context.Context, req *acce
 		if err != nil {
 			return nil, err
 		}
-		return upstreamDps.GetLatestBlockHeader(context, req)
+		ret, err := upstreamDps.GetLatestBlockHeader(context, req)
+		if err == nil {
```

Comment:
This should be != nil, otherwise errors thrown from GetLatestBlockHeader won't propagete

Reply:
Then we'll need another return to return 'ret, nil'

Reply:
Not anymore. I remember a latest discussion to try DPS first and fall back. We do not propagate the DPS error indeed, we still propagate the BDS one below. We can just propagate one.